docs/design/2018-07-19-row-format.md

```diff
@@ -0,0 +1,88 @@
+
+# Proposal: A new storage row format for efficient decoding
+
+- Author(s):     Yuxing ZHOU
+- Last updated:  2018-07-19
+
+## Abstract
+
+This proposal proposes a new row format for efficient decoding.
+
+## Background
+
+In the current row format, each column is encoded with the column ID followed by the column value and multiple columns are concatenated together. In order to find a specific column in the row, we need to start from the beginning and decode each column ID and the length of each value until we find the matched column. This decoding method is inefficient especially when the number of columns in a row is not small and the complexity is O(N).
+
+## Proposal
+
+The new format is defined as follows:
+
+![row-format](./row-format.jpeg)
+
+* 1 byte codec version
+
+    Its value starts from 128, so we can upgrade TiDB in a compatible way.
+
+* 1 byte row type flag containing the following attribute:
```

Comment:
I would suggest you replace "attribute" with "layout"

Reply:
It's more about attribute than layout.

Reply:
Gotcha